errtracker/errtracker.go

```diff
@@ -160,6 +168,104 @@ func detectVirt() string {
 	return strings.TrimSpace(string(output))
 }
 
+func journalError() string {
+	// TODO: look into using systemd package (needs refactor)
+	output, err := exec.Command("journalctl", "--utc", "--boot", "--priority=warning..err", "--lines=1000").CombinedOutput()
+	if err != nil {
+		if len(output) == 0 {
+			return fmt.Sprintf("error: %v", err)
+		}
+		output = append(output, fmt.Sprintf("\nerror: %v", err)...)
+	}
+	return string(output)
+}
+
+func procCpuinfoMinimal() string {
+	buf, err := ioutil.ReadFile(procCpuinfo)
+	if err != nil {
+		// if we can't read cpuinfo, we want to know _why_
+		return fmt.Sprintf("error: %v", err)
+	}
+	idx := bytes.LastIndex(buf, []byte("\nprocessor\t:"))
+	// if not found, return whole buffer; otherwise, return from just after the \n
+	return string(buf[idx+1:])
+}
+
+func procExe() string {
+	out, err := os.Readlink(procSelfExe)
+	if err != nil {
+		return fmt.Sprintf("error: %v", err)
+	}
+	return out
+}
+
+func procCwd() string {
+	out, err := os.Readlink(procSelfCwd)
+	if err != nil {
+		return fmt.Sprintf("error: %v", err)
+	}
+	return out
+}
+
+func procCmdline() string {
+	out, err := ioutil.ReadFile(procSelfCmdline)
+	if err != nil {
+		return fmt.Sprintf("error: %v", err)
+	}
+	return string(out)
+}
+
+func environ() string {
+	safeVars := []string{
+		"SHELL", "TERM", "LANGUAGE", "LANG", "LC_CTYPE",
+		"LC_COLLATE", "LC_TIME", "LC_NUMERIC",
+		"LC_MONETARY", "LC_MESSAGES", "LC_PAPER",
+		"LC_NAME", "LC_ADDRESS", "LC_TELEPHONE",
+		"LC_MEASUREMENT", "LC_IDENTIFICATION", "LOCPATH",
+	}
+	out := make([]string, 0, len(safeVars)+4)
```

Comment:
Why +4 here?

Reply:
I take it's for `PATH`, `LD_PRELOAD`, `LD_LIBRARY_PATH`, `XDG_RUNTIME_DIR`.